### PR TITLE
Create article domain, controller, service and domain classes

### PR DESCRIPTION
**Pull request by th0mr**
Added domain, controller, service and domain class for Article
Added an index.gsp page to display just the article names
Added tests for domain, controller and service classes

### CODE REVIEW

File: src/test/groovy/articleapp/ArticleControllerSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    def setup() {
+    }
+
+    def cleanup() {
+    }
+
+    void "Test index action returns nothing when no articles are present"() {
+        given:
+        Article testArticle2 = new Article(title:"testArticle2")
```
Comment (FergusMcGlynn) [inline src/test/groovy/articleapp/ArticleControllerSpec.groovy]: I don't think this line is necessary in this test - does it make any difference if it's removed?

**Review by FergusMcGlynn — COMMENTED** [review]
This looks really great! Nice and simple! I have left just one comment but nothing major.

File: src/test/groovy/articleapp/ArticleControllerSpec.groovy (hunk 1 above)
Comment (th0mr) [reply src/test/groovy/articleapp/ArticleControllerSpec.groovy]: Ah yes, good catch, that definitely shouldn't be there. Must have forgotten to remove it while messing around with the tests.

**Review by FergusMcGlynn — APPROVED** [review]
Great work! 🥇

**Pull request merged**